### ISSUE TITLE
use Rust AST parser to bypass JSON schema

### ISSUE THREAD

**Issue by pyramation** [pyramation/cosmwasm-typescript-gen#8]
https://github.com/RReverser/astexplorer-syn

**Comment by pyramation** [comment]
https://github.com/pyramation/rust-ast-stringify/blob/master/__tests__/index.test.js

**Comment by sascha1337** [comment]
so you kinda want to straight parse the sourcecode, not using the json schema, as that would be much more detailed ? nooow i get what your aim is here - let me PoC that out

**Comment by pyramation** [comment]
yes! we just need to pass in a string that represents rust source code - we don't need file system access. 

```ts
function (rustCode: string) => {
  let ast: JSON;
 /// some wasm magic

  return ast;
}
```

**Comment by sascha1337** [comment]
> yes! we just need to pass in a string that represents rust source code - we don't need file system access. 
> 
> 
> 
> ```ts
> 
> function (rustCode: string) => {
> 
>   let ast: JSON;
> 
>  /// some wasm magic
> 
> 
> 
>   return ast;
> 
> }
> 
> ```

can you inspect this AST representation of rust code below, and tell me, if this would be sufficient ?

ive got multiple solutions working, without wasi and without filesystem access - just need to place the bindings, which is not a huge thing, once i know, which AST 'syntax' is better to work with.

do you only care about rust=>ast json at the moment, or would the reverse logic ast to rs be helpful, for the current sprint ? if not, we can skip that and add to the backlog for later. 

lfg

**Comment by pyramation** [comment]
let's do deparser later (ast->rust)

cool! let me know when I can check something out :)

**Comment by sascha1337** [comment]
> > yes! we just need to pass in a string that represents rust source code - we don't need file system access. 
> > 
> > 
> > 
> > ```ts
> > 
> > function (rustCode: string) => {
> > 
> >   let ast: JSON;
> > 
> >  /// some wasm magic
> > 
> > 
> > 
> >   return ast;
> > 
> > }
> > 
> > ```
> 
> can you inspect this AST representation of rust code below, and tell me, if this would be sufficient ?
> 
> ive got multiple solutions working, without wasi and without filesystem access - just need to place the bindings, which is not a huge thing, once i know, which AST 'syntax' is better to work with.
> 
> do you only care about rust=>ast json at the moment, or would the reverse logic ast to rs be helpful, for the current sprint ? if not, we can skip that and add to the backlog for later. 
> 
> lfg 

Forgot to append the actual AST - upping asap
